src/WpfMath/TexFormulaParser.cs

```diff
@@ -572,16 +572,15 @@ internal static SourceSpan ReadElement(SourceSpan value, ref int position)
                 // Text style was found.
                 SkipWhiteSpace(value, ref position);
 
-                var styledFormula = command == TexUtilities.TextStyleName
-                    ? ConvertRawText(ReadElement(value, ref position), command)
-                    : Parse(ReadElement(value, ref position), command, environment.CreateChildEnvironment());
-
-                if (styledFormula.RootAtom == null)
-                    throw new TexParseException("Styled text can't be empty!");
-
-                var atom = AttachScripts(formula, value, ref position, styledFormula.RootAtom, true, environment);
-                var source = new SourceSpan(formulaSource.Source, formulaSource.Start, position);
-                formula.Add(atom, source);
+                var remainingString = ReadElement(value, ref position);
+                var remaining = command == TexUtilities.TextStyleName
+                    ? ConvertRawText(remainingString, command)
+                    : Parse(remainingString, command, environment.CreateChildEnvironment());
+
+                var source = value.Segment(start, position - start);
+                var styledAtom = new StyledAtom(source, remaining.RootAtom, null, null);
```

Comment:
I don't understand why we need this one. It is, um, a styled atom with no style added.

Reply:
Because it works! :)

What served to make this decision:
- the message in `throw new TexParseException("Styled text can't be empty!");`
- case similarity with `colorbox`

If I don’t add a wrapper, I will end up with a drop in the test [here (BoxTests.fs#L96)](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath.Tests/BoxTests.fs#L96).

Reply:
I understand why you've removed the `throw` statement, and that's good!

Why does that test break though?

Reply:
Because after `parse source` comes `null`

Reply:
Could you please clarify? I don't understand. It is here, in the local variable. How could it turn `null`?

Reply:
Because `TexFormula.RootAtom` is nullable. The situation is the same as, for example, with `colorbox` ([TexFormulaParser.cs#L480](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath/TexFormulaParser.cs#L480), [TexFormulaParser.cs#L591](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath/TexFormulaParser.cs#L591)).

Reply:
When the parsing of the formula in parentheses begins, the loops ([TexFormulaParser.cs#L217](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath/TexFormulaParser.cs#L217), [TexFormulaParser.cs#L293](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath/TexFormulaParser.cs#L293)) are skipped and `texFormula.RootAtom` remains set to `null`

Reply:
And, when we pass `atom.RootAtom` to the `Atom AttachScripts(...)` ([TexFormulaParser.cs#L627](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath/TexFormulaParser.cs#L627)), the result is `atom.RootAtom` ([TexFormulaParser.cs#L638](https://github.com/ForNeVeR/wpf-math/blob/7944747d6219f77b3f0f5dcd3a723d1c92a94dd5/src/WpfMath/TexFormulaParser.cs#L638)), i.e. `null`.

Reply:
And, sorry, but, I do not know what the `Atom AttachScripts(...)` should do, since I did not consider this further.